### ISSUE TITLE
install error

### ISSUE THREAD

**Issue by Orange15** [tensorflow/tensorflow#1452]
when i am installing from sources, there is an error like this:

ERROR: tensorflow/tensorflow/core/BUILD:98:1: //tensorflow/core:worker_service_proto_cc: no such attribute 'use_grpc_plugin' in 'cc_library' rule.
ERROR: tensorflow/tensorflow/core/BUILD:121:1: //tensorflow/core:master_service_proto_cc: no such attribute 'use_grpc_plugin' in 'cc_library' rule.
ERROR: tensorflow/tensorflow/cc/BUILD:61:1: Target '//tensorflow/core:tensorflow' contains an error and its package is in error and referenced by '//tensorflow/cc:tutorials_example_trainer'.
ERROR: Loading failed; build aborted.

**Comment by Ingwar** [comment]
I also got the similar error with fresh master:
```
ERROR: /home/igorkorsunov/tensorflow/tensorflow/core/BUILD:98:1: //tensorflow/core:worker_service_proto_cc: no such attribute 'use_grpc_plugin' in 'cc_library' rule.
ERROR: /home/igorkorsunov/tensorflow/tensorflow/core/BUILD:121:1: //tensorflow/core:master_service_proto_cc: no such attribute 'use_grpc_plugin' in 'cc_library' rule.
ERROR: /home/igorkorsunov/tensorflow/tensorflow/python/BUILD:154:1: Target '//tensorflow/core:protos_all_py' contains an error and its package is in error and referenced by '//tensorflow/python:framework_for_generated_wrappers'.
ERROR: Loading failed; build aborted.
```

I am using following command to build TensorFlow:
```
bazel build -c opt --config=cuda //tensorflow/tools/pip_package:build_pip_package 
```
Bazel version is 0.2.0

**Comment by girving** [comment]
@mrry: Have you seen this before?

**Comment by mrry** [comment]
Urgh, this sounds like `protobuf.bzl` might have changed... did we recently move to a newer version of protobuf? @keveman knows the mystic runes that make it work with gRPC.

**Comment by vrv** [comment]
I would also suggest maybe refreshing your submodules -- someone more git-competent than I can explain why git fetch/pull doesn't also update your submodules.

**Comment by vrv** [comment]
I thought that would be no different than resolving conflicts from the main repo :)

**Comment by LiberiFatali** [comment]
I got this kind of error, too

`ERROR: /home/hieunguyen/tensorflow/tensorflow/core/BUILD:98:1: //tensorflow/core:worker_service_proto_cc: no such attribute 'use_grpc_plugin' in 'cc_library' rule.
ERROR: /home/hieunguyen/tensorflow/tensorflow/core/BUILD:121:1: //tensorflow/core:master_service_proto_cc: no such attribute 'use_grpc_plugin' in 'cc_library' rule.
ERROR: /home/hieunguyen/tensorflow/tensorflow/cc/BUILD:61:1: Target '//tensorflow/core:tensorflow' contains an error and its package is in error and referenced by '//tensorflow/cc:tutorials_example_trainer'.
ERROR: Loading failed; build aborted.
INFO: Elapsed time: 0.072s`

when
`bazel build -c opt --config=cuda //tensorflow/cc:tutorials_example_trainer`

**Comment by LiberiFatali** [comment]
So what's the workaround for this?

**Comment by vrv** [comment]
Try:

```
git pull --recurse-submodules
git submodule update --recursive
```

**Comment by LiberiFatali** [comment]
Thanks for you reply.
I deleted everything, then, made a fresh clone of master source. Now the installation works well.

**Comment by sbrodeur** [comment]
I just had the same problem as mentioned above, and the solution proposed by @vrv fixed the issues for me.

**Comment by ssakhavi** [comment]
My problem was also solved via @vrv 's proposed fix.

**Comment by hrajanie** [comment]
I have been trying to install tensorflow on an EC2 instance based on [these instructions](http://eatcodeplay.com/installing-gpu-enabled-tensorflow-with-python-3-4-in-ec2/), and keep getting a similar error to the ones discussed above: 

`ERROR: /mnt/tmp/tensorflow/tensorflow/core/BUILD:87:1: //tensorflow/core:protos_all_py: no such attribute 'imports' in 'py_library' rule.
ERROR: /mnt/tmp/tensorflow/tensorflow/cc/BUILD:61:1: Target '//tensorflow/core:all_kernels' contains an error and its package is in error and referenced by '//tensorflow/cc:tutorials_example_trainer'.
ERROR: Loading failed; build aborted.`

Unfortunately, @vrv 's fix does not seem to work, and neither does retrying with a fresh master source. Any thoughts?

**Comment by hrajanie** [comment]
This was with 

`bazel build -c opt --config=cuda //tensorflow/cc:tutorials_example_trainer`

**Comment by vrv** [comment]
@davidzchen is this related to your change to remove python protobuf as a submodule?

**Comment by davidzchen** [comment]
The following error message seems to indicate that the version of Bazel being run does not support the `imports` attribute for the Python rules:

```
ERROR: /mnt/tmp/tensorflow/tensorflow/core/BUILD:87:1: //tensorflow/core:protos_all_py: no such attribute 'imports' in 'py_library' rule.
```

@hrajanie Which version of Bazel are you running? The `imports` attribute was added in 0.2.0.

**Comment by hrajanie** [comment]
@davidzchen Using Bezel 0.2.0 seems to have solved the problem I reported, although I'm still having problems building the Pip package. I may just try again using [these scripts](https://github.com/RealScout/deep-learning-images) instead. Thanks for your help!

**Comment by corcra** [comment]
I was also getting an error like this and fixed it by upgrading bazel from `0.1.4` to `0.2.3-2016-06-10` (installed bazel from source).

The specific error was:
```
ERROR: /tensorflow/tensorflow/contrib/session_bundle/BUILD:160:1: //tensorflow/contrib/session_bundle:manifest_proto_py: no such attribute 'imports' in 'py_library' rule.
ERROR: /tensorflow/tensorflow/tools/pip_package/BUILD:23:1: Target '//tensorflow/contrib/session_bundle:manifest_proto_py' contains an error and its package is in error and referenced by '//tensorflow/tools/pip_package:build_pip_package'.
ERROR: Loading failed; build aborted.
```
after running 
```bazel build -c opt --config=cuda //tensorflow/tools/pip_package:build_pip_package```

**Comment by tobegit3hub** [comment]
We have the same error like @hrajanie and @corcra in AWS EC2. 

And it works after upgrading bazel to 0.2.3 according to https://github.com/tensorflow/tensorflow/issues/2909
